docs/spec_draft.md

```diff
@@ -127,3 +127,183 @@ specification.
 %z = stablehlo.add %x, %y : tensor<2x2xf32>
 // %z: [[6, 8], [10, 12]]
 ```
+
+## stablehlo.and
+
+`stablehlo.and(lhs, rhs) -> result`
+
+### Semantics
+
+An element-wise operation performing bitwise AND of two tensors `lhs` and `rhs`
+of integer types and produces a `result` tensor. For Boolean tensors, it
+computes the logical operation.
+
+## Arguments
+
+| Name | Type |
+|-|-|
+| `lhs` | tensor of Integer or Boolean element types |
+| `rhs` | tensor of Integer or Boolean element types |
+
+## Results
+
+| Name | Type |
+|-|-|
+| `result` | tensor of Integer or Boolean element types |
+
+## Constraints
+
+  * `lhs`, `rhs` have the same type.
+  * Supported shapes: all static shapes.
+  * `result` must have the same type as that of `lhs` (or `rhs`).
+
+## Examples
+
+```
+// Bitwise operation with with Integer Tensors
+  // %x: [[1, 2], [3, 4]]
+  // %y: [[5, 6], [7, 8]]
+  %z = stablehlo.and %x, %y : tensor<2x2xsi32>
+  // %z: [[1, 2], [3, 0]]
+
+// Logical operation with with Boolean Tensors
+  // %x: [[false, false], [true, true]]
+  // %y: [[false, true], [false, true]]
+  %z = stablehlo.or %x, %y : tensor<2x2xpred>
+  // %z: [[false, false], [false, true]]
+```
+
+## stablehlo.not
+
+`stablehlo.not(operand) -> result`
+
+
+### Semantics
+
+An element-wise operation performing bitwise NOT of the given input tensor
+`operand` of type Integer and produces a `result` tensor. For Boolean
+tensors, it computes the logical NOT.
+
+### Arguments
+
+| Name | Type |
+|-|-|
+| `operand` | tensor of Integer or Boolean element types |
+
+
+### Results
+
+| Name | Type |
+|-|-|
+| `result` | tensor of Integer or Boolean element types |
+
+### Constraints
+
+  * Supported shapes: all static shapes.
```

Comment:
Missing constraint.

Reply:
Can you please point which one?

Reply:
Sorry! Same as above: "lhs, rhs have the same type".

Reply:
This being a unary op we can skip this constraint.